android-sdk/src/main/java/org/sagebionetworks/bridge/android/manager/dao/ConsentDAO.java

```diff
@@ -0,0 +1,95 @@
+package org.sagebionetworks.bridge.android.manager.dao;
+
+import android.content.Context;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import com.google.common.collect.Maps;
+import com.google.gson.reflect.TypeToken;
+
+import org.sagebionetworks.bridge.rest.model.ConsentSignature;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+import java.util.Set;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * Created by jyliu on 2/8/2017.
+ */
+public class ConsentDAO extends SharedPreferencesJsonDAO {
+    private static final Logger logger = LoggerFactory.getLogger(ConsentDAO.class);
+
+    private static final String PREFRENCES_FILE = "consents";
+    private static final String KEY_CONSENT_MAP = "consent-map";
+
+    private static final TypeToken<Map<String, ConsentSignature>> CONSENT_MAP_TYPE =
+            new TypeToken<Map<String, ConsentSignature>>() {
+            };
+
+
+    public ConsentDAO(Context applicationContext) {
+        super(applicationContext, PREFRENCES_FILE);
+    }
+
+    @NonNull
+    public Set<String> list() {
+        Set<String> subpopulations = load().keySet();
+
+        logger.debug("list called, found subpopulations: " + subpopulations);
+
+        return subpopulations;
+    }
+
+    @Nullable
+    public ConsentSignature get(@NonNull String subpopulationGuid) {
+        checkNotNull(subpopulationGuid);
+
+        Map<String, ConsentSignature> consents = load();
+        ConsentSignature consentSignature = consents.get(subpopulationGuid);
+
+        logger.debug("get called for subpopulation " + subpopulationGuid
+                + ", found: " + consentSignature);
+
+        return consentSignature;
+    }
+
+    public synchronized void put(@NonNull String subpopulationGuid,
```

Comment:
You can certainly store a map, but it's easier just to store the date in SharedPrefs with "subpopulationGuid" as the key, and the ConsentSignature as the value.  And then when you wanted the "map" at that point, you would just call sharedPrefs.getAll() which returns a Map<String, ?>